jobs/switchboard-leader/templates/readiness-switchboard.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/sh
+# Readiness probe script for the mysql-proxy role
+
+K=/var/vcap/packages/kubectl/bin/kubectl
+N="$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)"
+H="${HOSTNAME}"
+S="<%= p('gp.switchboard.service') %>"
+D="<%= p('gp.switchboard.renewal') %>"
+A=skiff-leader
+L=/tmp/log-ready-switchboard
+
+# Clear log.
+rm $L
+
+log () { echo "$@" >> $L ; }
+
+now () {
+    # timestamp, seconds of the epoch.
+    date '+%s'
+}
+
+get_claim () {
+    log retrieval ${N}:${S} :: $A
+    export CLAIM=$($K get service -n $N $S -o "jsonpath={.metadata.annotations.$A}")
+    # CLAIM='claimant:claimtime' -- claimtime unit is [epoch].
+    log claim: $CLAIM
+    test -n "${CLAIM}"
+}
+
+our_claim () {
+    CLAIMANT=$(echo $CLAIM | awk -F: '{ print $1 }')
+    log self? $H :: $CLAIMANT
+    test "${CLAIMANT}" == "${H}"
+}
+
+is_expired () {
+    CLAIMSEC=$(echo $CLAIM | awk -F: '{ print $2 }')
```

Comment:
Also known as `CLAIMSEC="${CLAIM#*:}"`

Reply:
Ok

Reply:
Done in commit 1202bd7fb7f61e3f8ab6450d714c36403293f1ea